### PR TITLE
add camel-casper component

### PR DESCRIPTION
**Pull request by abahmanem**

### CODE REVIEW

File: components/camel-casper/pom.xml [hunk 1]
```diff
@@ -0,0 +1,383 @@
...
+		</dependencies>
+	</dependencyManagement>
+
+	<dependencies>
+
+		<!-- https://mvnrepository.com/artifact/org.apache.commons/commons-lang3 -->
+		<dependency>
+			<groupId>org.apache.commons</groupId>
+			<artifactId>commons-lang3</artifactId>
+			<version>3.11</version>
```
Comment (orpiske) [inline components/camel-casper/pom.xml:77]: If possible / compatible should use `${commons-lang3-version}` to be aligned with the rest of the code base.

File: components/camel-casper/pom.xml [hunk 2]
```diff
@@ -0,0 +1,383 @@
...
+		</dependency>
+		<!-- https://mvnrepository.com/artifact/commons-cli/commons-cli -->
+		<dependency>
+			<groupId>commons-cli</groupId>
+			<artifactId>commons-cli</artifactId>
+			<version>1.4</version>
```
Comment (orpiske) [inline components/camel-casper/pom.xml:83]: If possible / compatible should use `${commons-cli-version}` to be aligned with the rest of the code base.

File: components/camel-casper/pom.xml [hunk 3]
```diff
@@ -0,0 +1,383 @@
...
+			<groupId>org.apache.camel</groupId>
+			<artifactId>camel-jsonpath</artifactId>
+			<version>${camel.version}</version>
+		</dependency>
+
+
+		<dependency>
+			<groupId>org.json</groupId>
+			<artifactId>json</artifactId>
+			<version>20211205</version>
```
Comment (orpiske) [inline components/camel-casper/pom.xml:115]: Versions should preferably be added to the parent and camel-dependencies pom for easier management on our side.

File: components/camel-casper/src/main/docs/casper-component.adoc [hunk 4]
```diff
@@ -0,0 +1,64 @@
+= Casper Camel Connector Component
+:doctitle: Casper Camel Connector
+:shortname: casper
+:artifactid: camel-casper
+:description: Camel casper endpoint : to interract with Casper nodes
+:since: 3.14.2
```
Comment (orpiske) [inline components/camel-casper/src/main/docs/casper-component.adoc:6]: Should be 3.20.0.

File: components/camel-casper/pom.xml [hunk 5]
```diff
@@ -0,0 +1,383 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (orpiske) [inline components/camel-casper/pom.xml:1]: Missing license header.

File: components/camel-casper/src/main/java/org/apache/camel/component/casper/CasperComponent.java [hunk 6]
```diff
@@ -0,0 +1,44 @@
+package org.apache.camel.component.casper;
```
Comment (orpiske) [inline components/camel-casper/src/main/java/org/apache/camel/component/casper/CasperComponent.java:1]: Missing license header.

File: components/camel-casper/src/main/java/org/apache/camel/component/casper/CasperConsumer.java [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+
+/**
+ * Camel CasperComsumer Component
+ * 
+ * @author mabahma
+ *
+ */
+
+public class CasperConsumer extends DefaultConsumer implements ShutdownAware, Suspendable {
+	public static final Logger logger = LoggerFactory.getLogger(CasperConsumer.class);
```
Comment (orpiske) [inline components/camel-casper/src/main/java/org/apache/camel/component/casper/CasperConsumer.java:26]: Checkstyle will catch this, but as a general comment: the `logger` variable needs to comply with the naming standards defined in our checkstyle rules. As such, I suggest naming it as `LOG`.

File: components/camel-casper/src/main/java/org/apache/camel/component/casper/CasperEventHandler.java [hunk 8]
```diff
@@ -0,0 +1,134 @@
...
+	/**
+	 * 
+	 */
+	@Override
+	public void onMessage(String evt, MessageEvent messageEvent) throws Exception {
+		JSONObject json = new JSONObject(messageEvent.getData());
+		String firstJsonPropertyKey = "";
+
+		if (json.keys().hasNext())
+			firstJsonPropertyKey = json.keys().next();
```
Comment (orpiske) [inline components/camel-casper/src/main/java/org/apache/camel/component/casper/CasperEventHandler.java:48]: Please use `{` and `}` for code blocks to comply with our coding standards defined in our checkstyle.

File: components/camel-casper/src/main/java/org/apache/camel/component/casper/examples/CasperProcessor.java [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+ */
+public class CasperProcessor implements Processor {
+
+	@Override
+	public void process(Exchange exchange) throws Exception {
+
+		Map<String, Object> map = exchange.getMessage().getHeaders();
+
+		for (String key : map.keySet()) {
+			System.out.println(key + ":" + map.get(key));
```
Comment (orpiske) [inline components/camel-casper/src/main/java/org/apache/camel/component/casper/examples/CasperProcessor.java:22]: This output should go to a logger.

File: components/camel-casper/src/test/java/org/apache/camel/component/casper/producer/CasperProducerWith_ACCOUNT_BALANCE_OperationTest.java [hunk 10]
```diff
@@ -0,0 +1,100 @@
...
+				"uref-9cC68775d07c211e44068D5dCc2cC28A67Cb582C3e239E83Bb0c3d067C4D0363-007");
+
+		template.send(exchange);
+		Exception exception = exchange.getException();
+		assertTrue(exception instanceof CamelExchangeException);
+		String expectedMessage = "stateRootHash parameter is required   with endpoint operation " + CasperConstants.ACCOUNT_BALANCE;
+		String actualMessage = exception.getMessage();
+
+		// assert Exception message
+		System.err.println(actualMessage+  "  "+expectedMessage.toLowerCase());
```
Comment (orpiske) [inline components/camel-casper/src/test/java/org/apache/camel/component/casper/producer/CasperProducerWith_ACCOUNT_BALANCE_OperationTest.java:84]: Output in tests should also go to a logger.

File: components/camel-casper/src/test/java/org/apache/camel/component/casper/producer/CasperProducerWith_BLOCK_TRANSFERS_OperationTest.java [hunk 11]
```diff
@@ -0,0 +1,70 @@
...
+		return false;
+	}
+
+	@Test
+	void testCallWithout_parameters() throws Exception {
+		Exchange exchange = createExchangeWithBodyAndHeader(null, CasperConstants.OPERATION, CasperConstants.BLOCK_TRANSFERS);
+		template.send(exchange);
+		Object body = exchange.getIn().getBody();
+		// assert Object is a List
+		assertTrue(body instanceof List);
```
Comment (orpiske) [inline components/camel-casper/src/test/java/org/apache/camel/component/casper/producer/CasperProducerWith_BLOCK_TRANSFERS_OperationTest.java:33]: I think you can use something like `assertIsInstance` (or a similar one) for this check.

File: components/camel-casper/src/test/java/org/apache/camel/component/casper/producer/CasperProducerWith_NETWORK_PEERS_OperationTest.java [hunk 12]
```diff
@@ -0,0 +1,50 @@
...
+		template.send(exchange);
+		Object body = exchange.getIn().getBody();
+		// assert Object is a List
+		assertTrue(body instanceof List);
+		List<PeerEntry> peers = (List<PeerEntry>) (body);
+		assertTrue(!peers.isEmpty());
+		// assert our List contains our node
+		//URI ourTestNode = new URI(CasperConstants.TESTNET_NODE_URL);
+		// assertTrue(peers.stream().anyMatch(s ->
+		// s.getAddress().substring(s.getAddress().indexOf(":")).equals(ourTestnode.getHost())));
```
Comment (orpiske) [inline components/camel-casper/src/test/java/org/apache/camel/component/casper/producer/CasperProducerWith_NETWORK_PEERS_OperationTest.java:38]: Commented code should be removed.

File: components/camel-casper/src/test/java/org/apache/camel/component/casper/consumer/sse/DataSetController.java [hunk 13]
```diff
@@ -0,0 +1,144 @@
...
+				emitter.completeWithError(e);
+			}
+		});
+		executor.shutdown();
+		return emitter;
+	}
+
+	private void randomDelay() {
+		try {
+			Thread.sleep(0);
```
Comment (orpiske) [inline components/camel-casper/src/test/java/org/apache/camel/component/casper/consumer/sse/DataSetController.java:139]: Please avoid using `Thread.sleep` as much as possible. Instead, please try to replace the code with Awaitility or, in case of synchronization / concurrency, latches and other concurrency handling mechanisms.

File: components/camel-casper/src/main/java/org/apache/camel/component/casper/examples/DemoApp.java [hunk 14]
```diff
@@ -0,0 +1,201 @@
...
+
+							}
+						});
+
+			}
+
+		});
+
+		cntxt.start();
+		Thread.sleep(5000);
```
Comment (orpiske) [inline components/camel-casper/src/main/java/org/apache/camel/component/casper/examples/DemoApp.java:87]: Please avoid using `Thread.sleep` as much as possible. Instead, please try to replace the code with Awaitility or, in case of synchronization / concurrency, latches and other concurrency handling mechanisms.

**Review by orpiske — CHANGES_REQUESTED** [review]
Thanks for your contribution. It has some problems that need to be fixed before it can be added to the code base.

File: components/camel-casper/pom.xml [hunk 15]
```diff
@@ -0,0 +1,383 @@
...
+			<name>mabahma</name>
+			<email>elmabahma@gmail.com</email>
+		</developer>
+		
+	</developers>
+	<scm>
+		<connection>scm:git:git@github.com:abahmanem/camel-casper.git</connection>
+		<developerConnection>scm:git:git@github.com:abahmanem/camel-casper.git</developerConnection>
+		<url>https://github.com/abahmanem/camel-casper</url>
+	</scm>
```
[the remark this answers is not in the document]
Comment (abahmanem) [reply components/camel-casper/pom.xml]: deleted this section

File: components/camel-casper/pom.xml (hunk 1 above)
Comment (abahmanem) [reply components/camel-casper/pom.xml]: put all these in properties as versions

File: components/camel-casper/src/main/docs/casper-component.adoc (hunk 4 above)
Comment (abahmanem) [reply components/camel-casper/src/main/docs/casper-component.adoc]: fixed

File: components/camel-casper/pom.xml (hunk 5 above)
Comment (abahmanem) [reply components/camel-casper/pom.xml:1]: fixed

File: components/camel-casper/src/main/java/org/apache/camel/component/casper/CasperComponent.java (hunk 6 above)
Comment (abahmanem) [reply components/camel-casper/src/main/java/org/apache/camel/component/casper/CasperComponent.java:1]: fixed

File: components/camel-casper/src/main/java/org/apache/camel/component/casper/CasperConsumer.java (hunk 7 above)
Comment (abahmanem) [reply components/camel-casper/src/main/java/org/apache/camel/component/casper/CasperConsumer.java]: fixed

File: components/camel-casper/src/main/java/org/apache/camel/component/casper/CasperEventHandler.java (hunk 8 above)
Comment (abahmanem) [reply components/camel-casper/src/main/java/org/apache/camel/component/casper/CasperEventHandler.java]: fixed

File: components/camel-casper/src/main/java/org/apache/camel/component/casper/examples/CasperProcessor.java (hunk 9 above)
Comment (abahmanem) [reply components/camel-casper/src/main/java/org/apache/camel/component/casper/examples/CasperProcessor.java]: fixed

File: components/camel-casper/src/test/java/org/apache/camel/component/casper/producer/CasperProducerWith_ACCOUNT_BALANCE_OperationTest.java (hunk 10 above)
Comment (abahmanem) [reply components/camel-casper/src/test/java/org/apache/camel/component/casper/producer/CasperProducerWith_ACCOUNT_BALANCE_OperationTest.java]: fixed

File: components/camel-casper/src/test/java/org/apache/camel/component/casper/producer/CasperProducerWith_BLOCK_TRANSFERS_OperationTest.java (hunk 11 above)
Comment (abahmanem) [reply components/camel-casper/src/test/java/org/apache/camel/component/casper/producer/CasperProducerWith_BLOCK_TRANSFERS_OperationTest.java]: fixed

File: components/camel-casper/src/main/docs/casper-component.adoc [hunk 16]
```diff
@@ -0,0 +1,64 @@
+= Casper Camel Connector Component
+:doctitle: Casper Camel Connector
+:shortname: casper
+:artifactid: camel-casper
+:description: Camel casper endpoint : to interract with Casper nodes
```
Comment (orpiske) [inline components/camel-casper/src/main/docs/casper-component.adoc:5]: I'd follow the description pattern used in other components. Please look at how we write the description for the `camel-kafka` component and try to describe this one on the same way.

File: components/camel-casper/src/main/docs/casper-component.adoc [hunk 17]
```diff
@@ -0,0 +1,64 @@
...
+:camel-spring-boot-name: casper
+
+*Since Camel {since}*
+
+*{component-header}*
+
+The Casper blockchain component uses the Casper Java SDK :
+https://github.com/syntifi/casper-sdk  and allows you to interact with Casper nodes.
+
+add the folllowing dependency maven  `pom.xml` file.
```
Comment (orpiske) [inline components/camel-casper/src/main/docs/casper-component.adoc:19]: Typo and missing capitalization.

File: components/camel-casper/src/main/java/org/apache/camel/component/casper/CasperConstants.java [hunk 18]
```diff
@@ -0,0 +1,73 @@
...
+
+/**
+ * Constants
+ * @author mabahma
+ */
+public interface CasperConstants {
+	//RPC Calls
+	String ENDPOINT_SERVICE = "NETWORK_PEERS, NODE_STATUS,DEPLOY,LAST_BLOCK,BLOCK,LAST_BLOCK_TRANSFERS,BLOCK_TRANSFERS,STATE_ROOT_HASH,ACCOUNT_INFO"
+            + ",AUCTION_INFO,ERA_INFO,STATE_ITEM,ACCOUNT_BALANC,ERPC_SCHEMA,GLOBAL_STATE,VALIDATOR_CHANGES,DICTIONARY_ITEM"
+            + "";
```
Comment (orpiske) [inline components/camel-casper/src/main/java/org/apache/camel/component/casper/CasperConstants.java:11]: I don't think this extra "" is necessary.

File: components/camel-casper/src/main/java/org/apache/camel/component/casper/CasperConstants.java [hunk 19]
```diff
@@ -0,0 +1,73 @@
+package org.apache.camel.component.casper;
+
+/**
+ * Constants
+ * @author mabahma
+ */
+public interface CasperConstants {
```
Comment (orpiske) [inline components/camel-casper/src/main/java/org/apache/camel/component/casper/CasperConstants.java:7]: Please use a class to define the constants instead of using an interface.

File: components/camel-casper/src/main/java/org/apache/camel/component/casper/CasperEndPoint.java [hunk 20]
```diff
@@ -0,0 +1,138 @@
...
+import com.syntifi.casper.sdk.service.CasperService;
+
+/**
+ * Camel casper endpoint : to interract with Casper nodes
+ * 
+ * @author p35862
+ *
+ */
+
+@UriEndpoint(firstVersion = "3.14.2", scheme = "casper", title = "Casper Camel Connector", syntax = "casper:nodeUrl", label = "casper", category = { Category.BITCOIN, Category.BLOCKCHAIN,
```
Comment (orpiske) [inline components/camel-casper/src/main/java/org/apache/camel/component/casper/CasperEndPoint.java:28]: The `firstVersion` is incorrect.

File: components/camel-casper/src/main/java/org/apache/camel/component/casper/CasperEndPoint.java [hunk 21]
```diff
@@ -0,0 +1,138 @@
...
+	public Consumer createConsumer(Processor processor) throws Exception {
+		URI uri = new URI(nodeUrl);
+		String event = configuration.getEvent();
+		if (!Arrays.asList(CasperConstants.CONSUMER_PATHS.split(",")).stream().anyMatch(s -> s.equals(uri.getPath())))
+			throw new InvalidPathException(uri.getPath(),
+					String.format("Invalid path '%s' for Casper Stream event server: expected '/events/main', '/events/deploys' or '/events/sigs ", uri.getPath()));
+		if (ConsumerEvent.findByName(event) != null) {
+			CasperConsumer consumer = new CasperConsumer(this, processor, configuration);
+			configureConsumer(consumer);
+			return consumer;
```
Comment (orpiske) [inline components/camel-casper/src/main/java/org/apache/camel/component/casper/CasperEndPoint.java:79]: This probably is going to be highlighted by checkstyle, but raising it nonetheless:  this formatting is incorrect due to missing `{`, `}` for the blocks.

File: components/camel-casper/src/main/java/org/apache/camel/component/casper/examples/DemoApp.java [hunk 22]
```diff
@@ -0,0 +1,204 @@
...
+import com.syntifi.casper.sdk.model.era.JsonEraValidators;
+import com.syntifi.casper.sdk.model.stateroothash.StateRootHashData;
+
+/**
+ * Demo Class loading Casper components example routes  
+ * @author p35862
+ *
+ */
+
+@SuppressWarnings("deprecation")
```
Comment (orpiske) [inline components/camel-casper/src/main/java/org/apache/camel/component/casper/examples/DemoApp.java:26]: Please don't suppress deprecation warnings. Ideally, they should be fixed prior to inclusion.

File: components/camel-casper/src/main/java/org/apache/camel/component/casper/examples/DemoApp.java [hunk 23]
```diff
@@ -0,0 +1,204 @@
...
+	/**
+	 * loads route 1
+	 * @param cntxt : Camel context
+	 * @param temp : producerTemplate
+	 * @throws Exception : exception
+	 */
+	private static void loadroute1(CamelContext cntxt, ProducerTemplate temp) throws Exception {
+		cntxt.addRoutes(new RouteBuilder() {
+			public void configure() throws Exception {
+				from("direct:" + CasperConstants.STATE_ROOT_HASH).routeId("STATE_ROOT_HASH")
```
Comment (orpiske) [inline components/camel-casper/src/main/java/org/apache/camel/component/casper/examples/DemoApp.java:39]: Nitpick: use `fromF` as it is cleaner than concatenating route code.

File: components/camel-casper/src/main/java/org/apache/camel/component/casper/examples/DemoApp.java [hunk 24]
```diff
@@ -0,0 +1,204 @@
...
+						.to("casper:http://65.21.227.180:7777/?operation=" + CasperConstants.STATE_ROOT_HASH)
+						.process(new Processor() {
+
+							@Override
+							public void process(Exchange exchange) throws Exception {
+								StateRootHashData state = (StateRootHashData) exchange.getIn().getBody();
+								LOG.info("* Current STATE_ROOT_HASH is : " + state.getStateRootHash());
```
Comment (orpiske) [inline components/camel-casper/src/main/java/org/apache/camel/component/casper/examples/DemoApp.java:46]: Please use log markers (ie.: `LOG.info("Current STATE_ROOT_HASH is : {}", state.getStateRootHash());` 

Also, as a general suggestion, avoid using custom markers/formatters in the log messages (i.e.: prefixing them with **** or other characters).

File: components/camel-casper/src/main/java/org/apache/camel/component/casper/examples/DemoApp.java [hunk 25]
```diff
@@ -0,0 +1,204 @@
...
+import com.syntifi.casper.sdk.model.stateroothash.StateRootHashData;
+
+/**
+ * Demo Class loading Casper components example routes  
+ * @author p35862
+ *
+ */
+
+@SuppressWarnings("deprecation")
+public class DemoApp {
```
Comment (orpiske) [inline components/camel-casper/src/main/java/org/apache/camel/component/casper/examples/DemoApp.java:27]: As this PR is very big already, I'd consider moving this as a separate PR to the [camel-examples repository](https://github.com/apache/camel-examples)

File: components/camel-casper/src/test/java/org/apache/camel/component/casper/consumer/CasperConsumerComponentTest.java [hunk 26]
```diff
@@ -0,0 +1,57 @@
+package org.apache.camel.component.casper.consumer;
+
+import static org.junit.Assert.assertThrows;
+//import static org.junit.jupiter.api.Assertions.assertThrows;
```
Comment (orpiske) [inline components/camel-casper/src/test/java/org/apache/camel/component/casper/consumer/CasperConsumerComponentTest.java:4]: Commented code should be removed.

File: components/camel-casper/src/test/java/org/apache/camel/component/casper/producer/CasperProducerWith_STATE_ITEM_OperationTest.java [hunk 27]
```diff
@@ -0,0 +1,88 @@
...
+import org.apache.camel.ProducerTemplate;
+import org.apache.camel.builder.RouteBuilder;
+import org.apache.camel.component.casper.CasperConstants;
+import org.apache.camel.component.casper.CasperTestSupport;
+import org.apache.commons.cli.MissingArgumentException;
+import org.junit.jupiter.api.Test;
+
+import com.syntifi.casper.sdk.model.storedvalue.StoredValue;
+
+@SuppressWarnings("rawtypes")
```
Comment (orpiske) [inline components/camel-casper/src/test/java/org/apache/camel/component/casper/producer/CasperProducerWith_STATE_ITEM_OperationTest.java:19]: Avoid suppressing warnings. Ideally, all warnings should be fixed prior to inclusion.

**Review by orpiske — CHANGES_REQUESTED** [review]
Thanks for the initial set of changes. I reviewed a bit more. 

There are a couple of problems that need to be fixed. 

1. I'd recommend ensuring the code complies with the coding standard. You can do so using `mvn -Pformat,fastinstall install` to format the code. Then, use `mvn -Psourcecheck` to ensure it passes through the checkstyle
2. There's files without the license header.

File: components/camel-casper/src/test/java/org/apache/camel/component/casper/producer/CasperProducerWith_NETWORK_PEERS_OperationTest.java (hunk 12 above)
Comment (abahmanem) [reply components/camel-casper/src/test/java/org/apache/camel/component/casper/producer/CasperProducerWith_NETWORK_PEERS_OperationTest.java]: fixed

File: components/camel-casper/pom.xml [hunk 28]
```diff
@@ -0,0 +1,395 @@
...
+		<artifactId>components</artifactId>
+		<version>3.20.0-SNAPSHOT</version>
+	</parent>
+
+
+	<artifactId>camel-casper</artifactId>
+	<packaging>jar</packaging>
+
+	<name>Camel :: Casper</name>
+	<description>Camel Casper blockchaine component</description>
```
[the remark this answers is not in the document]
Comment (abahmanem) [reply components/camel-casper/pom.xml:34]: fixed

File: components/camel-casper/pom.xml [hunk 29]
```diff
@@ -0,0 +1,395 @@
...
+	<url>https://casperlabs.io/</url>
+
+
+	<properties>
+		<compiler.source>1.8</compiler.source>
+		<compiler.target>1.8</compiler.target>
+		<maven-compiler-plugin.version>3.8.1</maven-compiler-plugin.version>
+		<maven-resources-plugin.version>3.2.0</maven-resources-plugin.version>
```
[the remark this answers is not in the document]
Comment (abahmanem) [reply components/camel-casper/pom.xml:42]: fixed

File: components/camel-casper/pom.xml [hunk 30]
```diff
@@ -0,0 +1,395 @@
...
+		<project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+		<project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
```
[the remark this answers is not in the document]
Comment (abahmanem) [reply components/camel-casper/pom.xml:44]: fixed

File: components/camel-casper/pom.xml [hunk 31]
```diff
@@ -0,0 +1,395 @@
...
+		<byte-buddy.version>1.9.3</byte-buddy.version>
+		<objenesis.version>2.6</objenesis.version>
+
+		<tomcat.version>9.0.54</tomcat.version>
+	</properties>
+
+	<licenses>
+		<license>
+			<name>MIT License</name>
+			<url>http://www.opensource.org/licenses/mit-license.php</url>
```
[the remark this answers is not in the document]
Comment (abahmanem) [reply components/camel-casper/pom.xml:68]: fixed

File: components/camel-casper/src/main/java/org/apache/camel/component/casper/examples/DemoApp.java (hunk 25 above)
Comment (abahmanem) [reply components/camel-casper/src/main/java/org/apache/camel/component/casper/examples/DemoApp.java:27]: fixed

File: components/camel-casper/src/test/java/org/apache/camel/component/casper/consumer/CasperConsumerComponentTest.java (hunk 26 above)
Comment (abahmanem) [reply components/camel-casper/src/test/java/org/apache/camel/component/casper/consumer/CasperConsumerComponentTest.java]: fixed

File: components/camel-casper/src/main/java/org/apache/camel/component/casper/examples/DemoApp.java (hunk 24 above)
Comment (abahmanem) [reply components/camel-casper/src/main/java/org/apache/camel/component/casper/examples/DemoApp.java]: fixed

File: components/camel-casper/src/test/java/org/apache/camel/component/casper/producer/CasperProducerWith_STATE_ITEM_OperationTest.java (hunk 27 above)
Comment (abahmanem) [reply components/camel-casper/src/test/java/org/apache/camel/component/casper/producer/CasperProducerWith_STATE_ITEM_OperationTest.java]: fixed

File: components/camel-casper/pom.xml [hunk 32]
```diff
@@ -0,0 +1,395 @@
...
+		<dependencies>
+			<dependency>
+				<groupId>org.apache.camel</groupId>
+				<artifactId>camel-bom</artifactId>
+				<version>${camel.version}</version>
+				<scope>import</scope>
+				<type>pom</type>
+			</dependency>
+		</dependencies>
+	</dependencyManagement>
```
[the remark this answers is not in the document]
Comment (abahmanem) [reply components/camel-casper/pom.xml:75]: fixed

File: components/camel-casper/pom.xml [hunk 33]
```diff
@@ -0,0 +1,395 @@
...
+			<artifactId>commons-lang3</artifactId>
+			<version>${commons-lang3.version}</version>
+		</dependency>
+		<dependency>
+			<groupId>commons-cli</groupId>
+			<artifactId>commons-cli</artifactId>
+			<version>${commons-cli.version}</version>
+		</dependency>
+
+		<!--//////////////////////////////////// Camel dependencies //////////////////////////////////// -->
```
[the remark this answers is not in the document]
Comment (abahmanem) [reply components/camel-casper/pom.xml:91]: fixed

File: components/camel-casper/src/main/java/org/apache/camel/component/casper/CasperConstants.java (hunk 19 above)
Comment (abahmanem) [reply components/camel-casper/src/main/java/org/apache/camel/component/casper/CasperConstants.java]: fixed

File: components/camel-casper/src/main/java/org/apache/camel/component/casper/CasperEndPoint.java (hunk 20 above)
Comment (abahmanem) [reply components/camel-casper/src/main/java/org/apache/camel/component/casper/CasperEndPoint.java]: changed

File: components/camel-casper/src/main/java/org/apache/camel/component/casper/examples/DemoApp.java (hunk 22 above)
Comment (abahmanem) [reply components/camel-casper/src/main/java/org/apache/camel/component/casper/examples/DemoApp.java:26]: fixed

File: components/camel-casper/src/main/java/org/apache/camel/component/casper/examples/DemoApp.java (hunk 23 above)
Comment (abahmanem) [reply components/camel-casper/src/main/java/org/apache/camel/component/casper/examples/DemoApp.java]: changed

File: components/camel-casper/pom.xml [hunk 34]
```diff
@@ -0,0 +1,290 @@
...
+		<dependency>
+			<groupId>org.apache.camel</groupId>
+			<artifactId>camel-jsonpath</artifactId>
+		</dependency>
+
+
+		<dependency>
+			<groupId>org.json</groupId>
+			<artifactId>json</artifactId>
+			<version>20211205</version>
```
Comment (oscerd) [inline components/camel-casper/pom.xml:104]: This kind of lib is using the json license, which is not compatible with Apache license. It should be removed and something else used.

File: components/camel-casper/src/main/docs/doc.md [hunk 35]
```diff
@@ -0,0 +1,259 @@
+# Camel casper producer component
+Camel Casper component for interacting with Casper network using RPC calls (Java SDK)
+
```
Comment (oscerd) [inline components/camel-casper/src/main/docs/doc.md:3]: The documentation file needs to be named like the component and need to be an .adoc, take a look at other components to copy the base skeleton

File: components/camel-casper/src/main/java/org/apache/camel/component/casper/CasperConfiguration.java [hunk 36]
```diff
@@ -0,0 +1,171 @@
+package org.apache.camel.component.casper;
```
Comment (oscerd) [inline components/camel-casper/src/main/java/org/apache/camel/component/casper/CasperConfiguration.java:1]: License missing.

**Review by oscerd — CHANGES_REQUESTED** [review]
There are some troubles with dependency licenses but also with license headers.

**Comment by oscerd** [comment]
Is this PR still active? @abahmanem

**Comment by abahmanem** [comment]
Yes,

I will work on it soon.

Thnx

Le mar. 22 nov. 2022 à 08:25, Andrea Cosentino ***@***.***> a
écrit :

**Comment by orpiske** [comment]
Is this still valid?

**Comment by orpiske** [comment]
@oscerd I think we can close this, as there hasn't been activity for a while. What do you think?

**Comment by oscerd** [comment]
yeah
